src/games/strategy/engine/framework/headlessGameServer/HeadlessServerMainPanel.java

```diff
@@ -115,7 +115,7 @@ private void addChat() {
     m_chatPanelHolder.removeAll();
     final ChatPanel chat;
     if (m_gameTypePanelModel != null && m_gameTypePanelModel.getPanel() != null && m_gameTypePanelModel.getPanel().getChatPanel() != null
-        && m_gameTypePanelModel.getPanel().getChatPanel() instanceof ChatPanel) {
+        && !m_gameTypePanelModel.getPanel().getChatPanel().isHeadless()) {
       chat = (ChatPanel) m_gameTypePanelModel.getPanel().getChatPanel();
```

Comment:
No. Wouldn't be a bad thing to spin up a headless server though and see what that process looks like. If you get time, would you mind jotting some notes down on that process in the wiki or pointing me to any that exist already? If I beat you to it I'll let you know where the wiki is. 

We may disagree, but I don't think that test should be a blocker to merging. Namely this code has been exercised enough that I'm confident it is okay (the non-headless case tested). Given what this change is, that has me confident. There are only two implementations of the ChatPanel interface, so either we are headless or not. So we switch from an "instanceof" check to an interface method, much more robust and less risky than an instanceof check that can silently fail if we make changes to the ChatPanel stuff.